lib/api/controller/document.js

```diff
@@ -559,6 +560,41 @@ class DocumentController extends NativeController {
     };
   }
 
+  /**
+  * Delete fields of a document. Throws an error if the document doesn't exist
+  *
+  * @param {Request} request
+  * @returns {Promise<Object>}
+  */
+  async deleteFields (request) {
+    const id = this.getId(request);
+    const fields = this.getBodyArray(request, 'fields');
+    const userId = this.getUserId(request);
+    const refresh = this.getString(request, 'refresh', 'false');
+    const { index, collection } = this.getIndexAndCollection(request);
+
+    const modifiedRequest = await global.kuzzle.validation.validate(request, false);
```

Comment:
The validation API need the entire document content. Here the body only contains the list of fields to delete.

Since we need to rewrite the validation API, maybe we can skip it for this action for now, what do you think @scottinet ?